tests/unit/test_logger.py

```diff
@@ -109,21 +107,26 @@ def test_nested_exceptions(self, caplog):
             except Exception as exc:
                 try:
                     raise DvcException("second") from exc
-                except DvcException:
-                    stack_trace = traceback.format_exc()
-                    logger.exception("message")
+                except DvcException as exc:
+                    try:
+                        raise ValueError("third") from exc
```

Comment:
Need to test how it works without explicit cause. Maybe that is even more important because that is an error in error handler presumably, which we want to fix.